### PR TITLE
added book, services, controllers

### PR DESCRIPTION
**Pull request by TarasHryts**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,32 @@
+
```
Comment (Sofasmile) [inline .gitignore]: a lot of redundant files

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,73 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-starter-parent</artifactId>
+        <version>2.2.0.RELEASE</version>
+        <relativePath/> <!-- lookup parent from repository -->
```
Comment (Sofasmile) [inline pom.xml]: remove coment

File: src/main/java/mate/academy/boot/hellobootdemo/controller/DataInitializer.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DataInitializer {
+    @Autowired
+    private BookService bookService;
+
+    @PostConstruct
+    public void initializeData() {
```
Comment (Sofasmile) [inline src/main/java/mate/academy/boot/hellobootdemo/controller/DataInitializer.java:15]: do we need two the same post construct?

File: src/main/java/mate/academy/boot/hellobootdemo/service/impl/BookServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+    @Override
+    public void delete(Long bookId) {
+        bookRepository.deleteById(bookId);
+    }
+
+    @Override
+    public Book update(Long bookId, Book book) {
+        Book bookToChange = bookRepository.findById(bookId).get();
+        bookToChange = book;
```
Comment (Sofasmile) [inline src/main/java/mate/academy/boot/hellobootdemo/service/impl/BookServiceImpl.java]: what do we do here? 
do we need to findById()?

File: src/main/resources/templates/books.html [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>all books</title>
+</head>
+<body>
+<h1>from all books</h1>
+</body>
+</html>
```
Comment (Sofasmile) [inline src/main/resources/templates/books.html]: add empty line

File: .gitignore (hunk 1 above)
Comment (TarasHryts) [reply .gitignore]: it is auto-generated .gitignore file

File: pom.xml (hunk 2 above)
Comment (TarasHryts) [reply pom.xml]: done

File: src/main/java/mate/academy/boot/hellobootdemo/controller/DataInitializer.java (hunk 3 above)
Comment (TarasHryts) [reply src/main/java/mate/academy/boot/hellobootdemo/controller/DataInitializer.java:15]: no, I delete it

File: src/main/java/mate/academy/boot/hellobootdemo/service/impl/BookServiceImpl.java (hunk 4 above)
Comment (TarasHryts) [reply src/main/java/mate/academy/boot/hellobootdemo/service/impl/BookServiceImpl.java]: changed

File: src/main/resources/templates/books.html (hunk 5 above)
Comment (TarasHryts) [reply src/main/resources/templates/books.html]: done

File: .gitignore (hunk 1 above)
Comment (Sofasmile) [reply .gitignore]: yes, but a lot of files we can remove

File: src/main/java/mate/academy/boot/hellobootdemo/controller/BookController.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+    @DeleteMapping("/{bookId}")
+    public void delete(@PathVariable("bookId") Long id) {
+        bookService.delete(id);
+    }
+
+    @PutMapping("/{bookId}")
+    public Book update(@PathVariable("bookId") Long bookId, BookDto bookDto) {
+        Book book = BookDtoUtil.createBookFromDto(bookDto);
+        bookService.update(bookId, book);
```
Comment (Sofasmile) [inline src/main/java/mate/academy/boot/hellobootdemo/controller/BookController.java]: which book should we pass here? dto or entity?

File: .gitignore (hunk 1 above)
Comment (TarasHryts) [reply .gitignore]: done

File: src/main/java/mate/academy/boot/hellobootdemo/controller/BookController.java (hunk 6 above)
Comment (TarasHryts) [reply src/main/java/mate/academy/boot/hellobootdemo/controller/BookController.java]: done

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**